pkg/target/regolib/src.rego

```diff
@@ -152,6 +154,51 @@ kind_matches(ks) {
   ks.kinds[_] == input.review.kind.kind
 }
 
+#######################
+# Name Selector Logic #
+#######################
+
+matches_name(match) {
+  not has_field(match, "name")
+}
+
+matches_name(match) {
+  has_field(match, "name")
+  input.review.name == match.name
+}
+
+matches_name(match) {
+  has_field(match, "name")
+  input.review.object.metadata.name == match.name
+}
+
+# oldObject covers Updates and Deletes.  The name of an object can't be changed, so there's no
+# need to worry about a difference between object and oldObject.
+matches_name(match) {
+  has_field(match, "name")
+  input.review.oldObject.metadata.name == match.name
+}
+
+matches_name(match) {
+  has_field(match, "name")
+  wildcard_name_match(match.name, input.review.name)
+}
+
+matches_name(match) {
+  has_field(match, "name")
+  wildcard_name_match(match.name, input.review.object.metadata.name)
+}
+
+matches_name(match) {
+  has_field(match, "name")
+  wildcard_name_match(match.name, input.review.oldObject.metadata.name)
+}
+
+wildcard_name_match(wildcard, subject) {
+  endswith(wildcard, "*")
+  glob.match(wildcard, [], subject)
```

Comment:
[glob.match](https://www.openpolicyagent.org/docs/latest/policy-reference/#glob) isn't strictly equivalent to prefixMatch. Is that intentional?

Reply:
IIRC we only allowed "*" to appear in the suffix, but looking now, I don't see where that's validated. @julianKatz Do you remember where that was enforced? Are we enforcing it for "name" as well?

Reply:
Looks like we forgot to add the JSON Schema validation for match criteria, we only have it for the config resource?

https://github.com/open-policy-agent/gatekeeper/pull/1404/files

I agree an explicit call for prefix matching would be less sensitive to these kinds of validation omissions.

Reply:
I'll add the same validation that we [used on namespaces](https://github.com/open-policy-agent/gatekeeper/blob/master/pkg/target/target.go#L42) to the `name` field.

@willbeason Will that provide safety against the edge case you have in mind?

`glob.match` _isn't_ quite the same, but I found this to be the easiest way to implement the prefixing when I [implemented the same feature](https://github.com/open-policy-agent/gatekeeper/pull/1404/files#diff-d0ed0e6a053bc28cdfcf4aff658c1e5d9ed5d5bd5b29cdc295e5734346bdd1c2R350) for namespaces.

Is there another way you'd suggest?

Reply:
I also realized that I'd already made a custom type ([PrefixWildcard](https://github.com/open-policy-agent/gatekeeper/blob/master/pkg/util/prefix_wildcard.go)) that I could use in all the prefix-matching golang code.  That gives me the pattern without having to copy and paste it again.  It also de-duplicates `prefixMatch` with the `Matches` function that already exists on that type.

@willbeason LMK if that looks good to you

Reply:
Nice!